<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compute scores from differential association analysis — dcScore • dcanr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compute scores from differential association analysis — dcScore"><meta property="og:description" content="Implementations and wrappers for existing implementations for
  methods inferring differential associations/co-expression. This method
  requires a matrix of expression and a binary condition to compute the
  differential association scores for all pairs of features (genes).
  Applications are not limited to analysis of gene expression data and may be
  used for differential associations in general."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dcanr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/dcanr_evaluation_vignette.html">Evaluating differential co-expression methods using dcanr</a>
    </li>
    <li>
      <a href="../articles/dcanr_vignette.html">Performing differential co-expression analysis using dcanr</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/DavisLaboratory/dcanr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute scores from differential association analysis</h1>
    <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/dcanr/blob/HEAD/R/inference_methods_generic.R" class="external-link"><code>R/inference_methods_generic.R</code></a></small>
    <div class="hidden name"><code>dcScore.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Implementations and wrappers for existing implementations for
  methods inferring differential associations/co-expression. This method
  requires a matrix of expression and a binary condition to compute the
  differential association scores for all pairs of features (genes).
  Applications are not limited to analysis of gene expression data and may be
  used for differential associations in general.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, <span class="va">dc.method</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for matrix</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for Matrix</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for data.frame</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for ExpressionSet</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for SummarizedExperiment</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for DGEList</span></span>
<span><span class="fu">dcScore</span><span class="op">(</span><span class="va">emat</span>, <span class="va">condition</span>, dc.method <span class="op">=</span> <span class="st">"zscore"</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>emat</dt>
<dd><p>a matrix, Matrix, data.frame, ExpressionSet, SummarizedExperiment
or DGEList</p></dd>


<dt>condition</dt>
<dd><p>a numeric, (with 1's and 2's representing a binary
condition), a factor with 2 levels or a character representing 2 conditions</p></dd>


<dt>dc.method</dt>
<dd><p>a character, representing the method to use. Use
<code><a href="dcMethods.html">dcMethods()</a></code> to get a list of methods</p></dd>


<dt>...</dt>
<dd><p>possible arguments are <code>cor.method</code>, <code>diffcoex.beta</code>,
<code>ebcoexpress.useBWMC</code>, <code>ebcoexpress.plot</code>, <code>ldgm.lambda</code>,
<code>ldgm.ntarget</code> and <code>ldgm.iter</code>. See details</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a matrix, of scores/statistics representing differential
  associations; p-values will be returned if FTGI is used and posterior
  probabilities if EBcoexpress is used.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>When using data from sequencing experiments, make sure appropriate
  filtering for low counts and data transformation has been performed. Not
  doing so will affect estimation of correlation coefficients which most
  methods rely on.</p>
<p>Additional method specific parameters can be supplied to the function.
  <code>cor.method</code> can be set to either 'pearson' (default) or 'spearman' to
  determine the method to use for estimating correlations. These are the two
  measures currently supported in the package. We recommend using the
  'spearman' correlation when dealing with sequencing data.</p>
<p>The beta parameter in the DiffCoEx method can be specified using
  <code>diffcoex.beta</code> (defaults to 6). This enable soft thresholding of
  correlations similar to WGCNA.</p>
<p>EBcoexpress specific parameters include <code>ebcoexpress.useBWMC</code>
  (defaults to <code>TRUE</code>) representing whether to use the bi-weight
  mid-correlation coefficient or not, and <code>ebcoexpress.plot</code> which plots
  the diagnostic plots if set to <code>TRUE</code> (defaults to <code>FALSE</code>).</p>
<p>LDGM specific parameters include <code>ldgm.lambda</code>, <code>ldgm.ntarget</code>
  and <code>ldgm.iter</code>. <code>ldgm.lambda</code> specifies the L1 regularisation
  parameter to use when fitting the model. This can be tuned and specified by
  the user. Alternatively, this can be tuned such that the resulting network
  has a specified number of edges. In this case, <code>ldgm.ntarget</code> should
  be specified instead. <code>ldgm.iter</code> is the maximum number of iterations
  to perform when tuning <code>ldgm.lambda</code> using <code>ldgm.ntarget</code>
  (defaults to 50).</p>
<p>EBcoexpress and GGM-based are implemented by providing interfaces to,
  or using functions from the <code>EBcoexpress</code>, <code>GeneNet</code>, and
  <code>COSINE</code> packages respectively. If using any of these methods, please
  <strong>cite</strong> the appropriate packages and the appropriate methodology
  articles.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="dcMethods.html">dcMethods</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">60</span><span class="op">)</span>, <span class="fl">2</span>, <span class="fl">30</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cond</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">15</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">dcScore</span><span class="op">(</span><span class="va">x</span>, <span class="va">cond</span><span class="op">)</span> <span class="co">#defaults to zscore</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            1          2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         NA -0.5583278</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 -0.5583278         NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"cor.method")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "pearson"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"call")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> z.score(emat = emat, condition = condition)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"dc.method")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "zscore"</span>
<span class="r-in"><span><span class="fu">dcScore</span><span class="op">(</span><span class="va">x</span>, <span class="va">cond</span>, dc.method <span class="op">=</span> <span class="st">'diffcoex'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             1           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          NA 0.000431787</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.000431787          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"cor.method")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "pearson"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"diffcoex.beta")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"call")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> diffcoex.score(emat = emat, condition = condition)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"dc.method")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "diffcoex"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Dharmesh D. Bhuva.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

