<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Performing differential co-expression analysis using dcanr • dcanr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Performing differential co-expression analysis using dcanr">
<meta property="og:description" content="dcanr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dcanr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dcanr_vignette.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/dcanr_vignette.html">Differential co-expression analysis using dcanr</a>
    </li>
    <li>
      <a href="../articles/dcanr_evaluation_vignette.html">DC method evaluation using dcanr</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/dcanr" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Performing differential co-expression analysis
using dcanr</h1>
                        <h4 data-toc-skip class="author">Dharmesh D
Bhuva</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052, AustraliaSchool of Mathematics
and Statistics, University of Melbourne, Parkville, VIC 3010,
Australia<br><h4 data-toc-skip class="date">2022-10-11</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/dcanr/blob/HEAD/vignettes/dcanr_vignette.Rmd" class="external-link"><code>vignettes/dcanr_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>dcanr_vignette.Rmd</code></div>

    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This package provides methods to perform differential co-expression
analysis and to evaluate differential co-expression methods using
simulated data. Differential co-expression analysis attempts to identify
gene-gene associations that change across conditions. Currently, 10
methods that identify changes between binary conditions are included: 8
are novel Bioconductor implementations of previously published methods,
and; 2 are accessed through interfaces to existing packages.</p>
<p>This vignette focuses on the application of differential
co-expression inference methods to real data. Available methods,
putative pipelines, and visualisations provided by the method are
introduced.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Download the package from Bioconductor</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"dcanr"</span><span class="op">)</span></span></code></pre></div>
<p>Or install the development version of the package from Github.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"DavisLaboratory/dcanr"</span><span class="op">)</span></span></code></pre></div>
<p>Load the installed package into an R session.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/dcanr/" class="external-link">dcanr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="available-inference-methods">Available inference methods<a class="anchor" aria-label="anchor" href="#available-inference-methods"></a>
</h2>
<p>The package implements 10 methods to infer differential co-expression
networks across binary conditions. The list of available methods can be
accessed by the <code><a href="../reference/dcMethods.html">dcMethods()</a></code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/dcanr/" class="external-link">dcanr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/dcMethods.html">dcMethods</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "dicer"       "diffcoex"    "ebcoexpress" "entropy"     "ftgi"       </span></span>
<span><span class="co">##  [6] "ggm-based"   "ldgm"        "magic"       "mindy"       "zscore"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="a-generic-differential-co-expression-analysis-pipeline">A generic differential co-expression analysis pipeline<a class="anchor" aria-label="anchor" href="#a-generic-differential-co-expression-analysis-pipeline"></a>
</h2>
<p>A differential co-expression pipeline generally consists of 4
steps:</p>
<ol style="list-style-type: decimal">
<li>Generating scores/statistics for each pair of genes</li>
<li>Assessing scores using statistical tests</li>
<li>Correcting for multiple hypothesis testing</li>
<li>Selecting differential associations</li>
</ol>
<p>Not all methods follow this pipeline. EBcoexpress computes posterior
probabilities therefore no statistical test needs to be performed and
steps 2-3 can be skipped. Like-wise DiffCoEx does not perform any
statistical tests and instead performs a soft-thresholding on the
scores. FTGI performs a statistical test and <span class="math inline">\(p\)</span>-values from this test are used as
scores, therefore step 2 is skipped. A standard analysis with the
z-score method using all 4 steps is shown here.</p>
</div>
<div class="section level2">
<h2 id="load-an-example-dataset-simulated">Load an example dataset (simulated)<a class="anchor" aria-label="anchor" href="#load-an-example-dataset-simulated"></a>
</h2>
<p>We first load an example simulated dataset (included in the package)
to extract the expression matrix and condition vector. Please note that
multiple knock-down experiments are performed per simulation and we use
one such knock-down as a condition here. The list of all knock-downs can
be retrieved using <code><a href="../reference/getSimData.html">getConditionNames()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sim102</span><span class="op">)</span></span>
<span><span class="co">#get available conditions</span></span>
<span><span class="fu"><a href="../reference/getSimData.html">getConditionNames</a></span><span class="op">(</span><span class="va">sim102</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ADR1" "UME6"</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#get expression data and conditions for 'UME6' knock-down</span></span>
<span><span class="va">simdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getSimData.html">getSimData</a></span><span class="op">(</span><span class="va">sim102</span>, cond.name <span class="op">=</span> <span class="st">'UME6'</span>, full <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">emat</span> <span class="op">&lt;-</span> <span class="va">simdata</span><span class="op">$</span><span class="va">emat</span></span>
<span><span class="va">ume6_kd</span> <span class="op">&lt;-</span> <span class="va">simdata</span><span class="op">$</span><span class="va">condition</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">emat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span> <span class="co">#149 genes and 406 samples</span></span></code></pre></div>
<pre><code><span><span class="co">##       sample_1  sample_2  sample_3  sample_4   sample_5</span></span>
<span><span class="co">## ADR1 0.4997864 0.4692072 0.3937255 0.1390989 0.05291159</span></span>
<span><span class="co">## FLO8 0.3340156 0.2845828 0.3682575 0.2263656 0.19292056</span></span>
<span><span class="co">## GIS1 0.3955383 0.4922471 0.5006614 0.2672858 0.48041978</span></span>
<span><span class="co">## IME4 0.4658504 0.3529325 0.4188025 0.3824827 0.33371681</span></span>
<span><span class="co">## KAR4 0.3705384 0.5812146 0.5282095 0.4945329 0.46677489</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ume6_kd</span><span class="op">)</span> <span class="co">#NOTE: binary conditions encoded with 1's and 2's</span></span></code></pre></div>
<pre><code><span><span class="co">## sample_1 sample_2 sample_3 sample_4 sample_5 sample_6 </span></span>
<span><span class="co">##        2        2        1        1        1        1</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="step-1-compute-scores">Step 1: Compute scores<a class="anchor" aria-label="anchor" href="#step-1-compute-scores"></a>
</h2>
<p>All inference methods can be accessed using the same call therefore
making it easier to change between methods. Method specific parameters
can be passed to this function and will be managed accordingly. The
default inference method is z-score therefore it does not need to be
specified via <code>dc.method</code>. We recommend using the Spearman
correlation as a measure of correlation as it is robust to outliers
which may be present in RNA-seq data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#apply the z-score method with Spearman correlations</span></span>
<span><span class="va">z_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dcScore.html">dcScore</a></span><span class="op">(</span><span class="va">emat</span>, <span class="va">ume6_kd</span>, dc.method <span class="op">=</span> <span class="st">'zscore'</span>, cor.method <span class="op">=</span> <span class="st">'spearman'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">z_scores</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##             ADR1        FLO8       GIS1       IME4       KAR4</span></span>
<span><span class="co">## ADR1          NA -0.05890109  1.0645602 -0.6869672 -0.2843683</span></span>
<span><span class="co">## FLO8 -0.05890109          NA  0.1855110  0.9256449  0.6803593</span></span>
<span><span class="co">## GIS1  1.06456016  0.18551097         NA  0.6336496 -1.2508845</span></span>
<span><span class="co">## IME4 -0.68696722  0.92564487  0.6336496         NA  0.4433642</span></span>
<span><span class="co">## KAR4 -0.28436825  0.68035927 -1.2508845  0.4433642         NA</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="step-2-perform-a-statistical-test">Step 2: Perform a statistical test<a class="anchor" aria-label="anchor" href="#step-2-perform-a-statistical-test"></a>
</h2>
<p>Appropriate statistical tests are automatically selected for the
method applied. Tests are applied on the result of the
<code><a href="../reference/dcScore.html">dcScore()</a></code> function (z-test for the z-score method and
permutation tests for other methods). The testing function returns the
score matrix (unmodified) if the method is either EBcoexpress, FTGI or
DiffCoEx.</p>
<p><em>NOTE: Do NOT modify the result of the scoring method as this will
result in failure of the testing function. This is intended as tests
should be performed for all computed scores to prevent bias in the
subsequent correction for multiple hypothesis testing. The same applies
for the next step.</em></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#perform a statistical test: the z-test is selected automatically</span></span>
<span><span class="va">raw_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dcTest.html">dcTest</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">emat</span>, <span class="va">ume6_kd</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">raw_p</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           ADR1      FLO8      GIS1      IME4      KAR4</span></span>
<span><span class="co">## ADR1        NA 0.9530309 0.2870750 0.4921034 0.7761282</span></span>
<span><span class="co">## FLO8 0.9530309        NA 0.8528283 0.3546306 0.4962770</span></span>
<span><span class="co">## GIS1 0.2870750 0.8528283        NA 0.5263095 0.2109766</span></span>
<span><span class="co">## IME4 0.4921034 0.3546306 0.5263095        NA 0.6575023</span></span>
<span><span class="co">## KAR4 0.7761282 0.4962770 0.2109766 0.6575023        NA</span></span></code></pre>
<p>For methods such as MINDy that require a permutation test, the number
of permutations can be specified by the <code>B</code> parameter.
Permutation tests are computationally expensive therefore we also
provide a parallelised implementation. See the help page of
<code>dcTest</code> for examples.</p>
</div>
<div class="section level2">
<h2 id="step-3-correcting-for-multiple-hypothesis-testing">Step 3: Correcting for multiple hypothesis testing<a class="anchor" aria-label="anchor" href="#step-3-correcting-for-multiple-hypothesis-testing"></a>
</h2>
<p>Since all pairwise combinations of genes are tested, <span class="math inline">\(p\)</span>-values need to be adjusted. Given <span class="math inline">\(n\)</span> genes, the total number of hypothesis
is <span class="math inline">\(\frac{n(n-1)}{2}\)</span> as the score
matrices are symmetric. Adjustment is performed accordingly. The default
adjustment function is <code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">stats::p.adjust</a></code> with the ‘fdr’
method used, however, custom functions and their parameters can be
specified instead. <code>dcAdjust</code> provides a wrapper to apply an
adjustment method to the raw <span class="math inline">\(p\)</span>-value matrix. Results from EBcoexpress
and DiffCoEx remain unmodified.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#adjust p-values (raw p-values from dcTest should NOT be modified)</span></span>
<span><span class="va">adj_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dcAdjust.html">dcAdjust</a></span><span class="op">(</span><span class="va">raw_p</span>, f <span class="op">=</span> <span class="va">p.adjust</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">adj_p</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           ADR1      FLO8      GIS1      IME4      KAR4</span></span>
<span><span class="co">## ADR1        NA 0.9933963 0.8879726 0.9289498 0.9655700</span></span>
<span><span class="co">## FLO8 0.9933963        NA 0.9748335 0.9116214 0.9289498</span></span>
<span><span class="co">## GIS1 0.8879726 0.9748335        NA 0.9328853 0.8494282</span></span>
<span><span class="co">## IME4 0.9289498 0.9116214 0.9328853        NA 0.9512398</span></span>
<span><span class="co">## KAR4 0.9655700 0.9289498 0.8494282 0.9512398        NA</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="step-4-generating-the-differential-co-expression-network">Step 4: Generating the differential co-expression network<a class="anchor" aria-label="anchor" href="#step-4-generating-the-differential-co-expression-network"></a>
</h2>
<p>The last step is thresholding the score/adjusted <span class="math inline">\(p\)</span>-value matrix to select differential
associations. Default adjusted <span class="math inline">\(p\)</span>-value thresholds of 0.1 are applied
where statistical tests are performed (to control for FDR at 0.1).
Results are presented as an <code>igraph</code> object shown below where
edges are coloured based on the score (negative to positive scores are
represented using the purple to green gradient of colours).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#get the differential network</span></span>
<span><span class="va">dcnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dcNetwork.html">dcNetwork</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">adj_p</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">dcnet</span>, vertex.label <span class="op">=</span> <span class="st">''</span><span class="op">)</span></span></code></pre></div>
<p><img src="dcanr_vignette_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert to an adjacency matrix</span></span>
<span><span class="va">adjmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/as_adjacency_matrix.html" class="external-link">as_adj</a></span><span class="op">(</span><span class="va">dcnet</span>, sparse <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">adjmat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      ADR1 FLO8 GIS1 IME4 KAR4</span></span>
<span><span class="co">## ADR1    0    0    0    0    0</span></span>
<span><span class="co">## FLO8    0    0    0    0    0</span></span>
<span><span class="co">## GIS1    0    0    0    0    0</span></span>
<span><span class="co">## IME4    0    0    0    0    0</span></span>
<span><span class="co">## KAR4    0    0    0    0    0</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert to a data.frame</span></span>
<span><span class="va">edgedf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html" class="external-link">as_data_frame</a></span><span class="op">(</span><span class="va">dcnet</span>, what <span class="op">=</span> <span class="st">'edges'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">edgedf</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    from   to     score     color</span></span>
<span><span class="co">## 1  ADR1 ACS1 -3.648617 #671F73B3</span></span>
<span><span class="co">## 2  ADR1 CTA1 -4.399843 #40004BB3</span></span>
<span><span class="co">## 3  ADR1 FOX2 -4.712234 #40004BB3</span></span>
<span><span class="co">## 4  ADR1 GUT1 -5.999899 #40004BB3</span></span>
<span><span class="co">## 5 STE12 BAR1  4.426577 #00441BB3</span></span>
<span><span class="co">## 6  SWI5 RME1 -3.780175 #5E186AB3</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] igraph_1.3.5     dcanr_1.13.4     BiocStyle_2.25.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] highr_0.9           RColorBrewer_1.1-3  bslib_0.4.0        </span></span>
<span><span class="co">##  [4] compiler_4.2.1      BiocManager_1.30.18 jquerylib_0.1.4    </span></span>
<span><span class="co">##  [7] iterators_1.0.14    tools_4.2.1         rngtools_1.5.2     </span></span>
<span><span class="co">## [10] digest_0.6.29       jsonlite_1.8.2      evaluate_0.16      </span></span>
<span><span class="co">## [13] memoise_2.0.1       pkgconfig_2.0.3     rlang_1.0.6        </span></span>
<span><span class="co">## [16] doRNG_1.8.2         foreach_1.5.2       cli_3.4.1          </span></span>
<span><span class="co">## [19] yaml_2.3.5          parallel_4.2.1      pkgdown_2.0.6.9000 </span></span>
<span><span class="co">## [22] xfun_0.33           fastmap_1.1.0       stringr_1.4.1      </span></span>
<span><span class="co">## [25] knitr_1.40          desc_1.4.2          fs_1.5.2           </span></span>
<span><span class="co">## [28] sass_0.4.2          GlobalOptions_0.1.2 systemfonts_1.0.4  </span></span>
<span><span class="co">## [31] grid_4.2.1          rprojroot_2.0.3     R6_2.5.1           </span></span>
<span><span class="co">## [34] textshaping_0.3.6   rmarkdown_2.16      bookdown_0.29      </span></span>
<span><span class="co">## [37] purrr_0.3.5         magrittr_2.0.3      codetools_0.2-18   </span></span>
<span><span class="co">## [40] htmltools_0.5.3     shape_1.4.6         colorspace_2.0-3   </span></span>
<span><span class="co">## [43] circlize_0.4.15     ragg_1.2.3          stringi_1.7.8      </span></span>
<span><span class="co">## [46] cachem_1.0.6</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Dharmesh D. Bhuva.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
